File: hw/carvajal/u2/hw19Inheritance/JSFarmSystem/model/Sheep.js
import { FarmAnimal } from "./FarmAnimal.js";

export class Sheep extends FarmAnimal {
    constructor(lastShearing, woolKg, id, breed, bornOn, gender, isAbleToReproduce, weight, cage) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        this.lastShearing = lastShearing;
        this.woolKg = woolKg;
    }

    cutWool() {
        this.woolKg = 0;
    }

    toString() {
        return `Sheep{lastShearing=${this.lastShearing}, woolKg=${this.woolKg}, ${super.toString()}}`;
    }
}

File: hw/chalan/u2/hw19Inheritance/Farmsystem/src/main/java/ec/edu/espe/farmsystem/model/Sheep.java
package ec.edu.espe.farmsystem.model;

import java.util.Date;

/**
 * @author Kevin Chalan, Object Masters, @ESPE
 */
public class Sheep extends FarmAnimal {
    
    private Date lastShearing;

    public Sheep(Date lastShearing, int id, String breed, Date bornOn, String gender, boolean isAbleToReproduce, float weight, Cage cage) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        this.lastShearing = lastShearing;
    }

    public void cutWool() {
        // Lógica para cortar lana
        System.out.println("Cutting wool for sheep " + getId());
        this.lastShearing = new Date(); // Actualizamos la fecha a 'hoy'
    }

    public void shear() {
        // Lógica de esquilar (similar a cutWool, según tu diagrama tienes ambos)
        cutWool();
    }

    @Override
    public String toString() {
        return "Sheep{" + "lastShearing=" + lastShearing + ", " + super.toString() + '}';
    }

    // Getters y Setters
    public Date getLastShearing() {
        return lastShearing;
    }

    public void setLastShearing(Date lastShearing) {
        this.lastShearing = lastShearing;
    }
}